### PR TITLE
Added a MHTML document loader

### PR DESCRIPTION
**Pull request by masylum**
MHTML is a very interesting format since it's used both for emails but also for archived webpages. Some scraping projects want to store pages in disk to process them later, mhtml is perfect for that use case.

This is heavily inspired from the beautifulsoup html loader, but extracting the html part from the mhtml file.

### CODE REVIEW

File: langchain/document_loaders/mhtml.py [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+                    if soup.title:
+                        title = str(soup.title.string)
+                    else:
+                        title = ""
+
+                    metadata: Dict[str, Union[str, None]] = {
+                        "source": self.file_path,
+                        "title": title,
+                    }
+                    return [Document(page_content=text, metadata=metadata)]
```
Comment (dev2049) [inline langchain/document_loaders/mhtml.py:68]: could there be more than one html part or will there always be at most one

Comment (masylum) [reply langchain/document_loaders/mhtml.py:68]: I don't think so. According to the spec:

> HTML [[RFC 1866](https://datatracker.ietf.org/doc/html/rfc1866)] defines a powerful means of specifying multimedia
   documents. These multimedia documents consist of a text/html root
   resource (object) and other subsidiary resources (image, video clip,
   applet, etc. objects) referenced by Uniform Resource Identifiers
   (URIs) within the text/html root resource. When an HTML multimedia
   document is retrieved by a browser, each of these component resources
   is individually retrieved in real time from a location, and using a
   protocol, specified by each URI.
   
   https://datatracker.ietf.org/doc/html/rfc2557

**Comment by rlancemartin** [comment]
Thanks for contributing! Please add an example notebook; you can see other example notebooks for loaders here as a guide: 

https://github.com/hwchase17/langchain/tree/master/docs/extras/modules/data_connection/document_loaders/integrations

**Comment by rlancemartin** [comment]
Also, please run `make format` to resolve the Lint errors.

**Comment by rlancemartin** [comment]
I added a notebook. Merging this now!

**Pull request merged**